tests/test_playwright.py

```diff
@@ -143,3 +143,38 @@ def check_colors():
                 expect(el).to_have_css("color", hover_color)
 
     _check_test_site(site_name, site_path, check_colors)
+
+
+def test_secondary_sidebar_toc_scrollspy(
+    sphinx_build_factory: Callable,
+    page: Page,
+    url_base: str,
+) -> None:
+    """Test that the secondary sidebar TOC highlights the correct item upon scroll."""
+    site_name = "colors"
+    site_path = _build_test_site(site_name, sphinx_build_factory=sphinx_build_factory)
+
+    def check_toc():
+        page.goto(urljoin(url_base, f"playwright_tests/{site_name}/index.html"))
+
+        toclinks = page.locator("#pst-page-toc-nav a.nav-link").all()
+        headings = []
+        for link in toclinks:
+            headings.append(page.locator(link.get_attribute("href")).first)
+
+        # Upon click, the first element should be active
+        toclinks[0].click()
+        expect(page.locator("p.copyright")).not_to_be_in_viewport()
+        assert "active" in toclinks[0].get_attribute("class")
+        page.locator("p.copyright").scroll_into_view_if_needed()
+
+        # After scrolling down, the first element shouldn't be active anymore
+        page.wait_for_function(
+            """() => !document
+                .querySelector('#pst-page-toc-nav a.nav-link')
+                .classList
+                .contains('active')
+            """
+        )
```

Comment:
To the extent possible, I think that since we opted for the Python Playwright API, we should stick with Python. Would the following work?

```py
        copyright = page.locator("p.copyright")
        copyright.scroll_into_view_if_needed()
        copyright.wait_for_element_state("visible")
        assert "active" not in toclinks[0].get_attribute("class")
```

I wonder if call to `wait_for_element_state()` is even necessary. Maybe it is, would have to test.

Reply:
So I thought about using `wait_for_element_state("visible")`, [but I don't think this will work](https://playwright.dev/python/docs/api/class-elementhandle#element-handle-wait-for-element-state):

> Element is considered visible when it has non-empty bounding box and does not have visibility:hidden computed style.
>
>Note that according to this definition:
>
>    Elements of zero size are not considered visible.
>    Elements with display:none are not considered visible.
>    Elements with opacity:0 are considered visible.

I think it is possible for an item to be outside the viewport but still technically visible. Am I reading that right?

Reply:
Sounds good to me - thanks!